### PR TITLE
Lambda Layer Update - OTel Java 1.11.1

### PR DESCRIPTION
**Pull request by vasireddy99**
This PR updates OTel Java dependencies to 1.11.1 and other dependencies to their latest available

It also updates the `_HANDLER` for the layer scripts with available [wrappers](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/aws-lambda/aws-lambda-events-2.2/library#using-wrappers)

### CODE REVIEW

File: java/dependencyManagement/build.gradle.kts [hunk 1]
```diff
@@ -9,17 +9,17 @@ plugins {
...
-    "software.amazon.awssdk:bom:2.17.112"
+    "software.amazon.awssdk:bom:2.17.132"
 )
 
 val DEPENDENCIES = listOf(
     "com.amazonaws:aws-lambda-java-core:1.2.1",
     "com.amazonaws:aws-lambda-java-events:3.11.0",
-    "com.squareup.okhttp3:okhttp:4.9.3",
-    "io.opentelemetry.javaagent:opentelemetry-javaagent:1.10.1"
+    "com.squareup.okhttp3:okhttp:5.0.0-alpha.4",
```
Comment (anuraaga) [inline java/dependencyManagement/build.gradle.kts]: `dependencyUpdates` is probably showing this but it should be kept at 4.9.3, the newest stable release

File: java/dependencyManagement/build.gradle.kts [hunk 2]
```diff
@@ -9,17 +9,17 @@ plugins {
 data class DependencySet(val group: String, val version: String, val modules: List<String>)
 
 val DEPENDENCY_BOMS = listOf(
-    "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.10.1-alpha",
-    "io.grpc:grpc-bom:1.42.1",
+    "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.11.0-alpha",
```
Comment (anuraaga) [inline java/dependencyManagement/build.gradle.kts]:
```suggestion
    "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.11.1-alpha",
```

File: java/dependencyManagement/build.gradle.kts [hunk 3]
```diff
@@ -9,17 +9,17 @@ plugins {
...
+    "software.amazon.awssdk:bom:2.17.132"
 )
 
 val DEPENDENCIES = listOf(
     "com.amazonaws:aws-lambda-java-core:1.2.1",
     "com.amazonaws:aws-lambda-java-events:3.11.0",
-    "com.squareup.okhttp3:okhttp:4.9.3",
-    "io.opentelemetry.javaagent:opentelemetry-javaagent:1.10.1"
+    "com.squareup.okhttp3:okhttp:5.0.0-alpha.4",
+    "io.opentelemetry.javaagent:opentelemetry-javaagent:1.11.0"
```
Comment (anuraaga) [inline java/dependencyManagement/build.gradle.kts]:
```suggestion
    "io.opentelemetry.javaagent:opentelemetry-javaagent:1.11.1"
```

File: java/layer-wrapper/build.gradle.kts [hunk 4]
```diff
@@ -10,7 +10,7 @@ dependencies {
     runtimeOnly(project(":awssdk-autoconfigure"))
 
     runtimeOnly("io.grpc:grpc-netty-shaded")
-    runtimeOnly("io.opentelemetry.instrumentation:opentelemetry-aws-lambda-1.0")
+    runtimeOnly("io.opentelemetry.instrumentation:opentelemetry-aws-lambda-1.0:1.10.1-alpha")
```
Comment (anuraaga) [inline java/layer-wrapper/build.gradle.kts]:
```suggestion
    runtimeOnly("io.opentelemetry.instrumentation:opentelemetry-aws-lambda-events-2.2")
```

Looks like our release notes missed mentioning this /cc @trask

Comment (trask) [reply java/layer-wrapper/build.gradle.kts]: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/5437

File: java/layer-wrapper/build.gradle.kts [hunk 5]
```diff
@@ -10,7 +10,7 @@ dependencies {
     runtimeOnly(project(":awssdk-autoconfigure"))
 
     runtimeOnly("io.grpc:grpc-netty-shaded")
```
Comment (anuraaga) [inline java/layer-wrapper/build.gradle.kts]: Ah has been true of a couple of releases but now would be a good time to remove gRPC dependencies as they aren't needed anymore. Should improve cold start time

File: java/dependencyManagement/build.gradle.kts [hunk 6]
```diff
@@ -9,17 +9,17 @@ plugins {
 data class DependencySet(val group: String, val version: String, val modules: List<String>)
 
 val DEPENDENCY_BOMS = listOf(
-    "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.10.1-alpha",
-    "io.grpc:grpc-bom:1.42.1",
+    "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.11.1-alpha",
+    "io.grpc:grpc-bom:1.44.0",
```
Comment (anuraaga) [inline java/dependencyManagement/build.gradle.kts]: Can remove this since unused now

**Review by anuraaga — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Comment by vasireddy99** [comment]
Thanks @anuraaga

**Review by NathanielRN — APPROVED** [review]
Nice! Love that it looks clean with removing those dependencies 😄

**Pull request merged**